### PR TITLE
Malf dynamic

### PR DESCRIPTION
**Pull request by Simplehorror**
# If this is your first PR, or not, take the time to read our CONTRIBUTING.md file! You can see it here: https://github.com/yogstation13/Yogstation/blob/master/.github/CONTRIBUTING.md

# Document the changes in your pull request
Malfunctioning AI now makes them malf instead of a traitor in dynamic. I also added a roundstart lawset for it as well as keeping traitor.
# Wiki Documentation
I don't think anything needs changing.

# Changelog

:cl:  
rscadd: Roundstart malfunctioning AI lawset
tweak: Midround malfunctioning AI now gives you malf, not traitor
/:cl:

### CODE REVIEW

**Review by ghost — APPROVED** [review]
When the AI is malf!

File: yogstation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			candidates -= player
+			continue
+		if(is_centcom_level(player.z))
+			candidates -= player
+			continue
+		if(player.mind && (player.mind.special_role || player.mind.antag_datums?.len > 0))
+			candidates -= player
+			continue
+		if(!(ROLE_MALF in player.client.prefs.be_special))
+			candidates -= player
```
Comment (Simplehorror) [inline yogstation/code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:33]: Actually, the malfunctioning AI midround rule doesn't have this, it starts removing living_players instead, seemed like a bug to me.

**Comment by Xoxeyos** [comment]
Don't shove this stuff into some modular folder, make the changes in the normal dynamic folder.

**Comment by Moja10** [comment]
Good change! Malf AI works best when paired with other antagonists, personally I have grown to dislike that malf is alone on its gamemode.

**Comment by Simplehorror** [comment]
> Don't shove this stuff into some modular folder, make the changes in the normal dynamic folder.

Why would I not modularise it

**Comment by Xoxeyos** [comment]
> > Don't shove this stuff into some modular folder, make the changes in the normal dynamic folder.
> 
> Why would I not modularise it

Because your changes are small in size, and I don't need dynamic scattered between more folders than it already is. 
Just keep it simple, and keep it in the main file path, outside of /yogstation/.

**Comment by Simplehorror** [comment]
> > > Don't shove this stuff into some modular folder, make the changes in the normal dynamic folder.
> > 
> > 
> > Why would I not modularise it
> 
> Because your changes are small in size, and I don't need dynamic scattered between more folders than it already is. Just keep it simple, and keep it in the main file path, outside of /yogstation/.

No I'm going to modularise it

**Comment by Xoxeyos** [comment]
> > > > Don't shove this stuff into some modular folder, make the changes in the normal dynamic folder.
> > > 
> > > 
> > > Why would I not modularise it
> > 
> > 
> > Because your changes are small in size, and I don't need dynamic scattered between more folders than it already is. Just keep it simple, and keep it in the main file path, outside of /yogstation/.
> 
> No I'm going to modularise it

Tell me why you think modularizing it is the best course of action.

**Comment by JamieD1** [comment]
Please just add it to the current dynamic rulesets. Im still undecided if this should be a thing.

**Comment by Simplehorror** [comment]
> Please just add it to the current dynamic rulesets. Im still undecided if this should be a thing.

You want me to remove the roundstart rule @JamieD1 ?

**Comment by JamieD1** [comment]
Sorry, Malf is just too powerfull to be in dynamic. Traitor is fine for now.

**Comment by Simplehorror** [comment]
> Sorry, Malf is just too powerfull to be in dynamic. Traitor is fine for now.

What threat would you place malf at?

**Pull request closed without merging**